packages/bundle-size/src/commands/uploadReport.js

```diff
@@ -0,0 +1,139 @@
+}
+
+/**
+ * @param {typeof import('../index') & { branch: string, 'commit-sha': string }} options
+ */
+async function uploadReport(options) {
+  if (!process.env.CI) {
+    console.log(`${chalk.red('[e]')} This is command can be executed only in CI`);
+    process.exit(1);
+  }
+
+  const { branch, 'commit-sha': commitSHA, quiet } = options;
+  const startTime = process.hrtime();
+
+  const localReportStartTime = process.hrtime();
+  const localReport = await collectLocalReport();
+
+  if (!quiet) {
+    console.log(
+      [chalk.blue('[i]'), `Local report prepared in ${hrToSeconds(process.hrtime(localReportStartTime))}`].join(' '),
+    );
+  }
+
+  if (typeof AZURE_ACCOUNT_KEY === 'undefined') {
+    console.log(
+      [
+        chalk.red('[e]'),
+        'process.env.BUNDLESIZE_ACCOUNT_KEY is not defined, please verify Azure Pipelines settings',
+      ].join(' '),
+    );
+    process.exit(1);
+  }
+
+  const credentials = new TablesSharedKeyCredential(AZURE_STORAGE_ACCOUNT, AZURE_ACCOUNT_KEY);
+  const client = new TableClient(
+    `https://${AZURE_STORAGE_ACCOUNT}.table.core.windows.net`,
+    AZURE_STORAGE_TABLE_NAME,
+    credentials,
+  );
+
+  const transaction = new TableTransaction();
+  const transactionStartTime = process.hrtime();
+
+  const entitiesIterator = await client.listEntities({
+    queryOptions: {
+      filter: odata`PartitionKey eq ${branch}`,
+    },
+  });
+
+  for await (const entity of entitiesIterator) {
+    // We can't delete and create entries with the same "rowKey" in the same transaction
+    // => we delete only entries not present in existing report
+    const shouldDelete = localReport.find(entry => createRowKey(entry) === entity.rowKey) === undefined;
+
+    if (shouldDelete) {
+      transaction.deleteEntity(/** @type {string} */ (entity.partitionKey), /** @type {string} */ (entity.rowKey));
+    }
+  }
+
+  localReport.forEach(entry => {
+    transaction.upsertEntity(
+      {
+        partitionKey: branch,
+        rowKey: createRowKey(entry),
+
+        name: entry.name,
+        packageName: entry.packageName,
+        path: entry.path,
+
+        minifiedSize: entry.minifiedSize,
+        gzippedSize: entry.gzippedSize,
+
+        commitSHA,
+      },
+      'Replace',
+    );
+  });
+
+  if (!quiet) {
+    console.log(
+      [chalk.blue('[i]'), `A transaction prepared in ${hrToSeconds(process.hrtime(transactionStartTime))}`].join(' '),
+    );
+  }
+
+  const submissionStartTime = process.hrtime();
+  await client.submitTransaction(transaction.actions);
+
+  if (!quiet) {
+    console.log(
+      [chalk.blue('[i]'), `A transaction submitted in ${hrToSeconds(process.hrtime(submissionStartTime))}`].join(' '),
+    );
+    console.log(`Completed in ${hrToSeconds(process.hrtime(startTime))}`);
+  }
+}
+
+// ---
+
+/** @type {import('yargs').CommandModule} */
+const api = {
+  command: 'upload-report',
+  describe: 'uploads local results to Azure Table Storage',
+  builder: function (yargs) {
```

Comment:
cant we make this static ? 

```
{
builder: {
    branch: {
      type: 'string',
      description: 'A branch to associate a report',
      required: true,
    },
    'commit-sha': {
      type: 'string',
      description: 'Defines a commit sha for a report',
      required: true,
    },
}
}
```

Reply:
Thanks for the suggestion, applied in 189e680.